controller/customdomain_controller_test.go

```diff
@@ -62,6 +69,82 @@ func TestCustomDomainController(t *testing.T) {
 		},
 	}
 
+	// A CustomDomain resource with routeSelector nil.
+	customdomainRouteSelectorNil := &customdomainv1alpha1.CustomDomain{
+		ObjectMeta: metav1.ObjectMeta{
+			Name:      instanceNameRouteSelectorNil,
+			Namespace: userNamespace,
+		},
+		Spec: customdomainv1alpha1.CustomDomainSpec{
+			Domain: userDomain,
+			Scope:  instanceScope,
+			Certificate: corev1.SecretReference{
+				Name:      userSecretName,
+				Namespace: userNamespace,
+			},
+			RouteSelector: &v1.LabelSelector{
+				MatchLabels: nil,
+			},
+		},
+	}
+
+	// A CustomDomain resource with routeSelector.
+	customdomainRouteSelector := &customdomainv1alpha1.CustomDomain{
+		ObjectMeta: metav1.ObjectMeta{
+			Name:      instanceNameRouteSelector,
+			Namespace: userNamespace,
+		},
+		Spec: customdomainv1alpha1.CustomDomainSpec{
+			Domain: userDomain,
+			Scope:  instanceScope,
+			Certificate: corev1.SecretReference{
+				Name:      userSecretName,
+				Namespace: userNamespace,
+			},
+			RouteSelector: &v1.LabelSelector{
+				MatchLabels: routeLabels,
+			},
+		},
+	}
+
+	// A CustomDomain resource with namespaceSelector nil.
+	customdomainNamespaceSelectorNil := &customdomainv1alpha1.CustomDomain{
+		ObjectMeta: metav1.ObjectMeta{
+			Name:      instanceNameNamespaceSelectorNil,
+			Namespace: userNamespace,
+		},
+		Spec: customdomainv1alpha1.CustomDomainSpec{
+			Domain: userDomain,
+			Scope:  instanceScope,
+			Certificate: corev1.SecretReference{
+				Name:      userSecretName,
+				Namespace: userNamespace,
+			},
+			NamespaceSelector: &v1.LabelSelector{
```

Comment:
I think you want to set NamespaceSelector to nil
```suggestion
			NamespaceSelector: nil
```